docs/content/en/docs/crd-ref/TaskDefinition/index.md

```diff
@@ -0,0 +1,279 @@
+---
+title: KeptnTaskDefinition
+description: Define tasks that can be run pre- or post-deployment
+weight: 89
+---
+
+A `KeptnTaskDefinition` defines tasks
+that are run by the Keptn Lifecycle Toolkit
+as part of the pre- and post-deployment phases of a `KeptnApp`.
+
+## Yaml Synopsis
+
+```yaml
+apiVersion: <lifecycle.keptn.sh/v?alpha?>
+kind: KeptnTaskDefinition
+metadata:
+  name: <task-name>
+spec:
+  function:
+    inline: | httpRef | functionRef
+    [parameters:
+      map:
+        textMessage: "This is my configuration"]
+    [secureParameters:
+      secret: slack-token]
+```
```

Comment:
Can do but wouldn't it be better to put that under "Examples"?  Also, I checked out the *keptntaskdefiniton.yaml files in that directory tree and they are sort of content-free at this time, with a #TODO to add fields.  Do we want to highlight those now or wait until they are populated?

Reply:
The `*keptntaskdefinitions` files have almost no content so I'm not sure I understand why we want to drive traffic to them -- isn't this just going to bring us a bunch of slack postings, annoy the users, and give other users the impression that something else is not done?  But I did add this link, under the "More examples" heading under `##Examples`.  I also added it to the `crd-template` file so it can be populated for subsequent pages.

Reply:
Then I would suggest to leave it for now and come back to it when we have more content done

Reply:
I was thinking the same thing.  Created issue: https://github.com/keptn/lifecycle-toolkit/issues/1143 .